### PR TITLE
Modifications to be compatible with remote storages

### PR DESCRIPTION
**Pull request by diox**

### CODE REVIEW

File: mkt/developers/tasks.py [hunk 1]
```diff
@@ -157,6 +157,10 @@ def _hash_file(fd):
 def resize_icon(src, dst, sizes, locally=False, **kw):
     """Resizes addon/websites icons."""
     log.info('[1@None] Resizing icon: %s' % dst)
+
+    open_ = open if locally else storage.open
+    delete = os.unlink if locally else storage.delete
+
```
Comment (washort) [inline mkt/developers/tasks.py]: nice

File: mkt/site/utils.py [hunk 2]
```diff
@@ -677,9 +678,16 @@ def get_icon_url(base_url_format, obj, size,
...
     else:
-        # [1] is the whole ID, [2] is the directory.
-        split_id = re.match(r'((\d*?)\d{1,3})$', str(obj.pk))
         # If we don't have the icon_hash set to a dummy string ("never"),
         # when the icon is eventually changed, icon_hash will be updated.
         suffix = obj.icon_hash or 'never'
+
+        if storage_is_remote():
+            # We don't care about base_url_format, the storage provides the url
+            # for a given path. We assume AWS_QUERYSTRING_AUTH is False atm.
```
Comment (washort) [inline mkt/site/utils.py]: let's add this to the config with a comment that turning it on breaks assumptions

**Comment by washort** [comment]
r+wc

File: mkt/site/utils.py (hunk 2 above)
Comment (diox) [inline mkt/site/utils.py]: Jason set that in settings already

**Pull request merged**